mlflow/store/artifact/runs_artifact_repo.py

```diff
@@ -132,63 +190,45 @@ def download_artifacts(self, artifact_path, dst_path=None):
-                    # TODO: Filter by 'source_run_id' once Databricks backend supports it
-                    filter_string=f"name = '{model_name}'",
-                    page_token=page_token,
-                )
-                for model in page:
-                    # Return the first model that matches the run_id and artifact_path
-                    if model.source_run_id == run_id:
-                        repo = get_artifact_repository(model.artifact_location)
-                        # TODO: Disabled for now. Consider re-enabling this once we migrate docs
-                        # and examples to use the new model URI format.
-                        # color_warning(
-                        #     "`runs:/<run_id>/artifact_path` is deprecated for loading models, "
-                        #     "use `models:/<model_id>` instead. Alternatively, retrieve "
-                        #     "`model_info.model_uri` from the model_info returned by "
-                        #     "mlflow.<flavor>.log_model. For example: "
-                        #     "model_info = mlflow.<flavor>.log_model(...); "
-                        #     "model = mlflow.<flavor>.load_model(model_info.model_uri)",
-                        #     stacklevel=1,
-                        #     color="yellow",
-                        # )
-                        return repo.download_artifacts(
-                            artifact_path=artifact_path,  # root directory
-                            dst_path=dst_path,
-                        )
 
-                if not page.token:
-                    break
-                page_token = page.token
+        model_out_path: Optional[str] = None
+        try:
+            model_out_path = self._download_model_artifacts(artifact_path, dst_path=dst_path)
+        except Exception:
             _logger.debug(
-                f"Failed to find any models with name {model_name} associated with the "
-                f"run {run_id}."
+                f"Failed to download model artifacts from {self.artifact_uri}/{artifact_path}.",
+                exc_info=True,
             )
-
-            raise MlflowException(
-                f"Failed to download artifacts from {full_path}. "
-                f"No model named {model_name!r} was found for run {run_id}. "
-                f"Please ensure that you've specified the correct model name or artifact path."
-            ) from e
+        return (
+            run_out_path
+            or model_out_path
+            or f"{dst_path}/{artifact_path.rsplit('/', maxsplit=1)[-1]}"
+        )
+
+    def _download_model_artifacts(self, artifact_path: str, dst_path: str) -> Optional[str]:
+        full_path = f"{self.artifact_uri}/{artifact_path}" if artifact_path else self.artifact_uri
+        run_id, rel_path = RunsArtifactRepository.parse_runs_uri(full_path)
+        if not rel_path:
+            # At least one part of the path must be present (e.g. "runs:/<run_id>/<name>")
```

Comment:
Reading this comment, it's not immediately obvious why we're dealing with a `runs:/` URI here. Can we add some more context to explain? Is there a better method name that might make this clearer?